### PR TITLE
Node 6, moved lib to src

### PR DESCRIPTION
**Pull request by rtsao**
- Use Node v6
- Rename  lib/ to src/
- Use default async to generator plugin
- Added necessary plugins for object rest/spread (see https://phabricator.babeljs.io/T7086)
- Update dependency semver strings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,27 +16,29 @@
...
+    "babel-cli": "6.x",
+    "babel-plugin-transform-async-to-generator": "6.x",
+    "babel-plugin-transform-es2015-destructuring": "6.x",
+    "babel-plugin-transform-es2015-parameters": "6.x",
+    "babel-plugin-transform-object-rest-spread": "6.x",
+    "babel-preset-es2015-node": "6.x",
     "body-parser": "1.15.0",
     "deep-extend": "0.4.1",
-    "express": "4.13.4",
+    "express": "^4.13.4",
```
Comment (scott113341) [inline package.json]: Do we want a loosey-goosey version specified?

**Comment by scott113341** [comment]
Other than above nitpick, LGTM!

File: package.json (hunk 1 above)
Comment (rtsao) [inline package.json]: Express is pretty enterprise at this point so I think we're pretty safe

Comment (scott113341) [inline package.json]: 🔒

Comment (rtsao) [inline package.json]: I figure if we actually care we should shrinkwrap, which I'm down for.

Comment (rtsao) [inline package.json]: https://github.com/L33T-KR3W/soundcasts-server/issues/33

Comment (scott113341) [inline package.json]: I think we should shrinkwrap, but that doesn't block this.  Merging.

**Pull request merged**